### PR TITLE
Call `super` in `API.inherited`

### PR DESCRIPTION
**Pull request by yogeshjain999**
`API.make_inheritable` was overriding `.inherited` hook on given instance itself. But it was not calling `super` which is required to bubble up the inheritance chain.

This PR removes `.make_inheritable` and calls `super` instead.

Fixes PR with reproducible spec #2179.

### CODE REVIEW

File: lib/grape/api.rb [hunk 1]
```diff
@@ -68,15 +69,6 @@ def call(*args, &block)
         instance_for_rack.call(*args, &block)
       end
 
-      # Allows an API to itself be inheritable:
-      def make_inheritable(api)
```
Comment (apotonick) [inline lib/grape/api.rb:72]: Why this change? Couldn't we just call `super` in `#inherited` and leave the rest as it is? :beers:

Comment (yogeshjain999) [reply lib/grape/api.rb:72]: Calling this method with `super` was creating warnings about method redefinations as `override_all_methods! ` gets called twice.

I guess `make_inheritable ` was defined in order to pass extra argument, instead of `super`.

**Comment by dblock** [comment]
Given that tests are passing I'm inclined to merge this. WDYT @apotonick (and thanks for jumping in to review!)?

**Comment by apotonick** [comment]
Now looking great! We need this `super` call to hook into your inheritance in [trb-endpoint](https://github.com/trailblazer/trailblazer-endpoint/), since we want to support the winning HTTP framework! :trophy: :beers:
